flang/lib/Lower/ConvertExpr.cpp

```diff
@@ -1705,14 +1705,20 @@ class ScalarExprLowering {
     return converter.genType(dt.GetDerivedTypeSpec());
   }
 
-  /// Apply the function `func` and return a reference to the resultant value.
-  /// This is required for lowering expressions such as `f1(f2(v))`.
+  /// Lower a function reference
   template <typename A>
-  ExtValue gen(const Fortran::evaluate::FunctionRef<A> &func) {
-    if (!func.GetType().has_value())
-      mlir::emitError(getLoc(), "internal: a function must have a type");
-    mlir::Type resTy = genType(*func.GetType());
-    ExtValue retVal = genProcedureRef(func, {resTy});
+  ExtValue genFunctionRef(const Fortran::evaluate::FunctionRef<A> &funcRef) {
+    if (!funcRef.GetType().has_value())
+      fir::emitFatalError(getLoc(), "internal: a function must have a type");
+    mlir::Type resTy = genType(*funcRef.GetType());
+    return genProcedureRef(funcRef, {resTy});
+  }
+
+  /// Lower function call `funcRef` and return a reference to the resultant
+  /// value. This is required for lowering expressions such as `f1(f2(v))`.
```

Comment:
It would be good to have a test for a case like `f1(f2(v))`.  I couldn't find one.

Reply:
I have added a test.